build.sbt

```diff
@@ -1,12 +1,18 @@
 import Dependencies._
 import sbtcrossproject.CrossPlugin.autoImport.crossProject
+def localSnapshotVersion = "2.2.3-SNAPSHOT"
```

Comment:
I am a bit concerned that this version needs to be bumped up every time we cut a new release. It would be nice to avoid that.

Reply:
We can parse it from git tags

```scala
def parseTagVersion: String = {
    import scala.sys.process._
    "git describe --abbrev=0 --tags".!!.drop(1).strip
  }

def localSnapshotVersion: String = s"$parseTagVersion-SNAPSHOT"
```

Reply:
Ah, sorry I just realized that it(my local published build with `2.1.0+xxx-SNAPSHOT`) happened just because I haven't fetched tags from a remote repository for a while... oh my ignorance...

> I am a bit concerned that this version needs to be bumped up every time we cut a new release. It would be nice to avoid that.

Now, I agree with that. 👍 

---

> We can parse it from git tags

Awesome! With this `localSnapshotVersion`, we can simplify the regex in `ScalafmtDynamicDownloader` 👍

Reply:
May be document [here](https://github.com/scalameta/scalafmt/blob/master/docs/contributing-scalafmt.md) that it is recommended to do `git fetch` before local building?

Reply:
It might be helpful though it won't be a problem for most people who are new to this project :)
BTW, since that documentation is becoming older and things have changed a bit (like release process, document locations), we have to update it in another PR.